File: daq/slc/src/base/src/Module.cc
#include "Module.h"

#include "Host.h"

#include "StringUtil.h"

using namespace Belle2;

const std::string Module::getSQLFields() const throw()
{
  return Node::getSQLFields() + ", product_id int, location text";
}

const std::string Module::getSQLLabels() const throw()
{
  return Node::getSQLLabels() + ", product_id, location";
}

const std::string Module::getSQLValues() const throw()
{
  return Node::getSQLValues() +
         Belle2::form(", %d, '%s'", getProductID(),
                      getLocation().c_str());
}

File: daq/slc/tools/javalib/b2rc/src/b2rc/core/Module.java
package b2rc.core;

public abstract class Module extends SCObject {

	private int _product_id;
	private String _location;

	public Module() {
		_product_id = -1;
	}

	public int getProductID() {
		return _product_id;
	}

	public void setProductID(int id) {
		_product_id = id;
	}

	public String getLocation() {
		return _location;
	}

	public void setLocation(String location) {
		_location = location;
	}

	@Override
	public String getSQLFields() {
		return super.getSQLFields() + ", product_id int, location text";
	}

	@Override
	public String getSQLLabels() {
		return super.getSQLLabels() + ", product_id, location";
	}

	@Override
	public String getSQLValues() {
		return super.getSQLValues() +  ", " + _product_id + ", '" + _location + "'";
	}

};
